samples/NativeCatalog/android/app/src/main/java/com/pspdfkit/nativecatalog/CustomPdfViewManager.java

```diff
@@ -0,0 +1,309 @@
+        }
+    }
+
+    @NonNull
+    @Override
+    public String getName() {
+        return "CustomPdfView";
+    }
+
+    @NonNull
+    @Override
+    protected PdfView createViewInstance(@NonNull ThemedReactContext reactContext) {
+        Activity currentActivity = reactContext.getCurrentActivity();
+        if (currentActivity instanceof FragmentActivity) {
+            FragmentActivity fragmentActivity = (FragmentActivity) currentActivity;
+
+            // You can create a PdfView in your own view manager and use all the work our wrapper already does.
+            PdfView pdfView = new PdfView(reactContext);
+
+            // Make sure to inject all the classes the view needs to function.
+            pdfView.inject(fragmentActivity.getSupportFragmentManager(),
+                reactContext.getNativeModule(UIManagerModule.class).getEventDispatcher());
+
+            // Setup any configuration you want, you could even make this a prop.
+            PdfActivityConfiguration config = new PdfActivityConfiguration.Builder(reactContext)
+                .setSignaturePickerOrientation(SignaturePickerOrientation.UNLOCKED)
+                .signatureSavingStrategy(SignatureSavingStrategy.NEVER_SAVE)
+                .autosaveEnabled(false)
+                .build();
+            pdfView.setConfiguration(config);
+
+            // The pdf view needs a fragment tag to function, make sure this is unique inside your app.
+            pdfView.setFragmentTag("pdfView");
+
+            // Finally return it to react-native.
+            return pdfView;
+        }
+
+        throw new IllegalStateException("CustomPdfView can only be used in FragmentActivity subclasses.");
+    }
+
+    // This will allow us to set the document via react-native.
+    @ReactProp(name = "document")
+    public void setDocument(PdfView view, @NonNull String document) {
+        view.setDocument(document);
+    }
+
+    @Nullable
+    @Override
+    public Map getExportedCustomDirectEventTypeConstants() {
+        // These are required for the PdfView to work.
+        // You can see them in ReactPdfViewManager.java
+        Map map = MapBuilder.of(PdfViewStateChangedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onStateChanged"),
+            PdfViewDocumentSavedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onDocumentSaved"),
+            PdfViewAnnotationTappedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onAnnotationTapped"),
+            PdfViewAnnotationChangedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onAnnotationsChanged"),
+            PdfViewDataReturnedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onDataReturned"),
+            PdfViewDocumentSaveFailedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onDocumentSaveFailed"),
+            PdfViewDocumentLoadFailedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onDocumentLoadFailed")
+        );
```

Comment:
Could we extract these registrations to the PdfView class, so that a single call is sufficient to set them up? Something like:

```java
Map map = PdfView.defaultEventRegistrationMap();
```